src/scipp/core/data_group.py

```diff
@@ -128,7 +129,8 @@ def __getitem__(self, name):
                 f"Positional indexing dim '{dim}' not possible as the length is not "
                 "unique.")
         return DataGroup({
-            key: var[dim, index] if dim in _item_dims(var) else var
+            key: var[dim, index] if
+            (isinstance(var, Bins) or dim in _item_dims(var)) else var
```

Comment:
So now this raises an exception if an items is bins but does not have the slice dim?

Reply:
No. It is fine to not have the dim, but it will raise if the coord is not found.

Reply:
Ah ok.

Reply:
I am actually not happy with it, since it raises exceptions from different places, depending on what you do wrong (such as passing no `dim`, passing integers instead of variables, ...).